### PR TITLE
Add support for displaying the version with `wt --version`

### LINKED ISSUE #5494: Unclear how to determine installed version?
For me, it's unclear how to determine what my currently installed version of the Windows Terminal is. I tried the following things:

**Check the command-line arguments**: but it's not listed, and "probing" for `--version` gives an error that it's not an argument that's expected.

**Check the Windows Store** (I installed it through that), but I don't see a version anywhere. I don't really use the Windows Store, so maybe I overlooked it, but these are the places I checked:
 * the detailed listing
 * My Library > Installed: it states when modified, but not the version
 * My Library > Installed: same as above
 * My Library > Downloads: it states versions, but not the Windows Terminal is not listed under Recent Activity, but Available Updates, and that doesn't list a version (see Context below) 

**Looked for an issue about this** in this repo, but couldn't find anything related: almost any issue has the word version in it, since it's asked when creating an issue 😅 

**Checked the issue templates**, although there is listed how to find the version of Windows, not of the Windows Terminal

**The properties pane on the `wt.exe` executables** I could find, but they're completely empty, as the files are also 0 bytes. I checked these locations:
 * `C:\Users\ties\AppData\Local\Microsoft\WindowsApps\wt.exe`
 * `C:\Users\ties\AppData\Local\Microsoft\WindowsApps\Microsoft.WindowsTerminal_8wekyb3d8bbwe`

# Context
I was looking into an issue I keep running into (#3609), and noticed I was running an older version of the Windows Terminal. Before I was going to update it, I wanted to known which version I have installed, so I can cross check all the related issues, and comments, if I can expect it to be fixed.

# Suggested Improvements
 * Add to the issue templates
 * Add to the main docs somewhere
 * Add support for `--version` argument to `wt.exe`

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request
Here's 3000 words:

image
image
image

## References
* #4632 

## PR Checklist
* [x] Closes #5494
* [x] I work here
* [ ] Tests added/passed
* [n/a] Requires documentation to be updated

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -1631,6 +1631,8 @@ namespace winrt::TerminalApp::implementation
     //   or 0. (see TerminalPage::_ParseArgs)
     int32_t TerminalPage::SetStartupCommandline(winrt::array_view<const hstring> args)
     {
+        _appArgs.SetVersionString(ApplicationDisplayName(), ApplicationVersion());
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1634]: I'm a bit alarmed about this. Can AppArgs load up the static AppLogic singleton and cut out the middleman (and the setter in the idl, and the hstrings, etc.)?

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/TerminalPage.cpp:1634]: (like the settings serializer does)

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: oh yea I forgot about that

File: src/cascadia/TerminalApp/AppCommandlineArgs.cpp [hunk 2]
```diff
@@ -151,6 +157,22 @@ int AppCommandlineArgs::_handleExit(const CLI::App& command, const CLI::Error& e
 // - <none>
 void AppCommandlineArgs::_buildParser()
 {
+    auto versionCallback = [this](int64_t /*count*/) {
+        if (const auto appLogic{ winrt::TerminalApp::implementation::AppLogic::Current() })
+        {
+            // Set our message to display the application name and the current version.
+            _exitMessage = fmt::format("{0}\n{1}",
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/AppCommandlineArgs.cpp:164]: yyyyaaaasssss

File: src/cascadia/WindowsTerminal/AppHost.cpp [hunk 3]
```diff
@@ -115,7 +115,10 @@ void AppHost::_HandleCommandlineArgs()
                             GetStringResource(messageTitle).data(),
                             MB_OK | messageIcon);
 
-                ExitProcess(result);
+                if (_logic.ShouldExitEarly())
```
Comment (DHowett-MSFT) [inline src/cascadia/WindowsTerminal/AppHost.cpp:118]: It looks like we're still covered in the `version` case by the `message` being non-empty -- we could simplify this changeset more if there's no compelling reason to introduce ShoudlExitEarly up through the stack. What do you think?

Comment (zadjii-msft) [reply src/cascadia/WindowsTerminal/AppHost.cpp:118]: So sure, we don't _need_ it in this PR anymore. Originally I was only going to exit early if there were no subcommands specified after a `--version`.

Now, I'm thinking about using it in the future for `open-settings` - if there's no subcommands other than `open-settings`, just open the file and exit. I don't feel strongly about keeping it around for now, but I will probably be using it in the future so ¯\\\_(ツ)\_/¯

**Comment by WSLUser** [comment]
Can we merge this in? If it's got enough approvals, we should get this in for v1.

**Pull request merged**